### PR TITLE
Add README-RIOT.rst

### PR DESCRIPTION
**Pull request by kYc0o**
This PR adds a first README for the usage of MCUBoot with RIOT. It's a very basic version, inspired from the README for Zephyr.

### CODE REVIEW

File: README-RIOT.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+which should be able to boot any of the supported OS images.
+
+Building Applications for the bootloader
+========================================
+
+A compatible mcuboot image can be compiled by typing: :code:`make mcuboot`.
+
+The only variable which needs to be set is :code:`IMAGE_VERSION` loaded
+with a valid formatted value. The format is :code:`major.minor.patch+other`
+(e.g. :code:`export IMAGE_VERSION= 1.1.1+1`. This varibale can be either
```
Comment (utzig) [inline README-RIOT.rst]: typo `variable`

File: README-RIOT.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Signing the application
+-----------------------
+
+The application will be automatically signed with the provided key.
+If no key is provided, a new key will be automatically generated. The
+default key type is RSA-2048.
+
+In order to use your provided key, you need to recompile the bootloader
+using you public key, either in Zephyr or mynewt by following the
```
Comment (utzig) [inline README-RIOT.rst]: Mynewt capitalized!

**Comment by utzig** [comment]
Btw, commits need to be signed off.

**Comment by kYc0o** [comment]
Squashed and signed-off added.

**Pull request merged**